src/components/application_manager/test/commands/mobile/system_request_test.cc

```diff
@@ -114,18 +126,237 @@ class SystemRequestTest
                             am::commands::Command::CommandOrigin::ORIGIN_SDL));
   }
 
+  void Run(MessageSharedPtr message) {
+    SharedPtr<SystemRequest> command(CreateCommand<SystemRequest>(message));
+    ASSERT_TRUE(command->Init());
+    command->Run();
+  }
+
+  void SetUp() OVERRIDE {
+    file_system::CreateDirectoryRecursively(binary_data_folder);
+    file_system::CreateDirectoryRecursively(kSystemFilesPath + "/");
+    file_system::CreateFile(app_full_file_path);
+    file_system::CreateFile(kSystemFilesPath + "/" + kFileName);
+  }
+
+  void TearDown() OVERRIDE {
+    file_system::RemoveDirectory(kAppStorageFolder);
+    file_system::RemoveDirectory(kSystemFilesPath);
+  }
+
   MockAppPtr mock_app_;
   MockPolicyHandlerInterface mock_policy_handler_;
+  const std::string binary_data_folder;
+  const std::string app_full_file_path;
+  std::vector<uint8_t> binary_data_;
 };
 
 TEST_F(SystemRequestTest, Run_HTTP_FileName_no_binary_data_REJECTED) {
   PreConditions();
-  MessageSharedPtr msg = CreateIVSUMessage();
+
+  EXPECT_CALL(mock_policy_handler_,
+              IsRequestTypeAllowed(kAppPolicyId, kRequestHttpType))
+      .WillOnce(Return(true));
 
   ExpectManageMobileCommandWithResultCode(mobile_apis::Result::REJECTED);
 
-  SharedPtr<SystemRequest> command(CreateCommand<SystemRequest>(msg));
-  command->Run();
+  Run(CreateIVSUMessage());
+}
+
+TEST_F(SystemRequestTest,
+       Run_HttpRequestTypeAndExistsFileName_SuccessResponce) {
```

Comment:
@BSolonenko propose to rename to `Run_HttpRequestTypeAndFileExists_SuccessResponse`